backend/pkg/boards/blcu.go

```diff
@@ -0,0 +1,235 @@
+package boards
+
+import (
+	"bytes"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/network/tftp"
+	dataPacket "github.com/HyperloopUPV-H8/h9-backend/pkg/transport/packet/data"
+	"time"
+)
+
+// TODO! Get from ADE
+const (
+	BlcuName = "BLCU"
+	BlcuId   = abstraction.BoardId(1)
+
+	AckId      = "1"
+	DownloadId = "2"
+	UploadId   = "3"
+
+	BlcuOrderId = 1
+
+	DownloadName = "download"
+	UploadName   = "upload"
+
+	DSuccess = "download success"
+	USuccess = "upload success"
+)
+
+type BLCU struct {
+	api     abstraction.BoardAPI
+	ackChan chan struct{}
+	ip      string
+}
+
+func New(ip string) *BLCU {
+	return &BLCU{
+		ackChan: make(chan struct{}),
+		ip:      ip,
+	}
+}
+func (boards *BLCU) Id() abstraction.BoardId {
+	return BlcuId
+}
+
+func (boards *BLCU) Notify(notification abstraction.BoardNotification) {
+	switch notification := notification.(type) {
+	case AckNotification:
+		boards.ackChan <- struct{}{}
+
+	case DownloadEvent:
+		err := boards.download(notification)
+		if err != nil {
+			ErrDownloadFailure{
+				Timestamp: time.Now(),
+				Inner:     err,
+			}.Error()
+		}
+	case UploadEvent:
+		err := boards.upload(notification)
+		if err != nil {
+			ErrDownloadFailure{
+				Timestamp: time.Now(),
+				Inner:     err,
+			}.Error()
+		}
+	default:
+		ErrInvalidBoardEvent{
+			Event:     notification.Event(),
+			Timestamp: time.Now(),
+		}.Error()
+	}
+}
+
+func (boards *BLCU) SetAPI(api abstraction.BoardAPI) {
+	boards.api = api
+}
+
+func (boards *BLCU) download(notification abstraction.BoardNotification) error {
+	// Notify the BLCU
+
+	// TODO! Ask for message procedure
+	ping := dataPacket.NewPacketWithValues(abstraction.PacketId(BlcuOrderId),
+		make(map[dataPacket.ValueName]dataPacket.Value),
```

Comment:
This map should have a value which indicates the board where the code will be downloaded/uploaded (the board value comes from the notification)

Reply:
also remember to set the other map the value of the same name to true

Reply:
We have to look at this, as I need to catch what you mean here.